mapviz/src/mapviz.cpp

```diff
@@ -201,7 +209,23 @@ void Mapviz::Initialize()
       connect(&spin_timer_, SIGNAL(timeout()), this, SLOT(SpinOnce()));
     }
 
-    node_ = new ros::NodeHandle("mapviz");
+    node_ = new ros::NodeHandle("~");
```

Comment:
> But before we merge this in, something to consider: should existing ImageTransport subscriptions automatically re-subscribe using the new transport if it's changed?

>This would require adding a mechanism to notify plugins that they should re-subscribe to their topics.

I think it should automatically resubscribe, and I can imagine there will be future use-cases for that interface.